language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -904,3 +947,31 @@ fn str_escape(s: &str) -> Result<String, fmt::Error> {
 fn path_escape(p: &Path) -> Result<String, fmt::Error> {
     str_escape(p.to_str().ok_or(fmt::Error)?)
 }
+
+/// Checks if dependencies of a given package in two different dependency graph maps are the
+/// same.
+fn pkg_deps_equal(
+    pkg_name: Symbol,
+    pkg_graph: &DiGraphMap<PM::PackageName, Dependency>,
+    other_graph: &DiGraphMap<PM::PackageName, Dependency>,
+) -> (Vec<Symbol>, Vec<Symbol>) {
+    let pkg_edges = BTreeSet::from_iter(pkg_graph.edges(pkg_name).map(|(_, pkg, dep)| (dep, pkg)));
+    let other_edges =
+        BTreeSet::from_iter(other_graph.edges(pkg_name).map(|(_, pkg, dep)| (dep, pkg)));
+
+    let intersection = BTreeSet::from_iter(pkg_edges.intersection(&other_edges).cloned());
+    if intersection.len() == pkg_edges.len() && intersection.len() == other_edges.len() {
+        return (vec![], vec![]);
+    }
+    let pkg_deps = pkg_edges
+        .difference(&intersection)
+        .cloned()
+        .map(|(_, pkg)| pkg)
+        .collect();
+    let other_deps = other_edges
+        .difference(&intersection)
+        .cloned()
+        .map(|(_, pkg)| pkg)
+        .collect();
+    (pkg_deps, other_deps)
```

Comment:
nit: Might be neater to use `symmetric_difference` and `partition`?

```suggestion
    let pkg_edges = BTreeSet::from_iter(pkg_graph.edges(pkg_name).map(|(_, pkg, dep)| (dep, pkg));
    let other_edges = 
        BTreeSet::from_iter(other_graph.edges(pkg_name).map(|(_, pkg, dep)| (dep, pkg);
    
    pkg_edges
        .symmetric_difference(other_edges)
        .partition(|dep| pkg_edges.contain(dep));
```

Also, I would recommend returning the whole dependency (both package name and the dependency), because the mismatch could be in the `Dependency` part, in which case we want to avoid a confusing error message where it says there's a mismatch but shows no mismatched packages.